examples/dbscan/reader.cpp

```diff
@@ -76,11 +78,184 @@ std::vector<ArborX::Point> loadArborXData(std::string const &filename,
+  // that and assuming 2) The second line indicates total trajectories in file
+  // (I'm ignoring that and observing how many are there by reading them). All
+  // lines that follow contains a trajectory separated by new line. The first
+  // number in the trajectory is the number of points followed by location points
+  // separated by spaces
+
+  std::vector<float> longitudes;
+  std::vector<float> latitudes;
+  std::vector<ArborX::Point> v;
+
+  int lineNo = -1;
+  int wordNo = 0;
+  int lonlatno = 100;
+
+  float thisWord;
+  while (fgets(line, sizeof(line), fp_data))
+  {
+    // std::cout << line << std::endl;
+    if (lineNo > -1)
+    {
+      char *pch;
+      char *end_str;
+      wordNo = 0;
+      lonlatno = 0;
+      pch = strtok_r(line, "\"[", &end_str);
+      while (pch != nullptr)
+      {
+        if (wordNo > 0)
+        {
+          char *pch2;
+          char *end_str2;
+
+          pch2 = strtok_r(pch, ",", &end_str2);
+
+          if (strcmp(pch2, "]") < 0 && lonlatno < 255)
+          {
+
+            thisWord = atof(pch2);
+
+            if (thisWord != 0.00000)
+            {
+              if (thisWord > -9 && thisWord < -7)
+              {
+                longitudes.push_back(thisWord);
+                // printf("lon %f",thisWord);
+                pch2 = strtok_r(nullptr, ",", &end_str2);
+                thisWord = atof(pch2);
+                if (thisWord < 42 && thisWord > 40)
+                {
+                  latitudes.push_back(thisWord);
+                  // printf(" lat %f\n",thisWord);
+
+                  lonlatno++;
+                }
+                else
+                {
+                  longitudes.pop_back();
+                }
+              }
+            }
+          }
+        }
+        pch = strtok_r(nullptr, "[", &end_str);
+        wordNo++;
+      }
+      // printf("num lonlat were %d x 2\n",lonlatno);
+    }
+    lineNo++;
+    if (lonlatno <= 0)
+    {
+      lineNo--;
+    }
+
+    // printf("Line %d\n",lineNo);
+  }
+  fclose(fp_data);
+
+  int num_points = longitudes.size();
+  assert(longitudes.size() == latitudes.size());
+  v.reserve(num_points);
+  for (unsigned int i = 0; i < num_points; ++i)
+    v.emplace_back(longitudes[i], latitudes[i], 0.f);
+
+  std::cout << "done\nRead in " << v.size() << " points" << std::endl;
+
+  return v;
+}
+
+std::vector<ArborX::Point> load3dSpatialNetworkData(std::string const &filename,
+                                                    bool binary)
+{
+  assert(!binary);
+  std::cout << "Trying to open " << filename
+            << " assuming 3dSpatialNetwork data.\n";
+  std::ifstream file(filename);
+  assert(file.good());
+
+  std::vector<ArborX::Point> v;
+
+  std::string thisWord;
+  while (file.good())
+  {
+    getline(file, thisWord, ',');
+    getline(file, thisWord, ',');
+    float longitude = stof(thisWord);
+    getline(file, thisWord, ',');
+    float latitude = stof(thisWord);
```

Comment:
I'm not sure I understand this. The file has four fields separated by commas, right? What happens with the forth one?

The other thing that I worry about is that I just tried to use something similar on a file with three fields "a,b,c", and calling getline on the last field with ',' results in it catching the beginning of the next line. Have you double checked that this produces the exact numbers it is supposed to?

Reply:
Wow, that's really a case on cancelling errors 😮 The *only* reason it works is because they ignore the first and the last field. So the first `getline` calls catches the end of the previous line *and* the beginning of the next line 🤯 And it did not work for me because I actually care about the first field.